bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -630,6 +643,23 @@ public TableInfo materializeViewToTable(
     return materializeTable(querySql, tableId, expirationTimeInMinutes);
   }
 
+  public Schema getTableSchema(String querySql, Map<String, String> additionalQueryJobLabels) {
+    JobInfo jobInfo =
+        JobInfo.of(
+            jobConfigurationFactory
+                .createQueryJobConfigurationBuilder(querySql, additionalQueryJobLabels)
+                .setDryRun(true)
+                .build());
+
+    log.info("running query {}", querySql);
```

Comment:
Can log "dry running query" to differentiate between actual query run and this dry run.